awx/main/tests/unit/test_tasks.py

```diff
@@ -246,6 +246,8 @@ def status_side_effect(pk, **kwargs):
             # If `Job.update_model` is called, we're not actually persisting
             # to the database; just update the status, which is usually
             # the update we care about for testing purposes
+            if kwargs.get('result_traceback'):
+                raise Exception('Task encountered error:\n{}'.format(kwargs['result_traceback']))
```

Comment:
@ryanpetrello I hate to put in drive-by fixes/enhancements like this, but within the course of doing these features, I've hit this a solid handful of times.

### before

```
_______________________________________________________________ TestInventoryUpdateCredentials.test_tower_source[True] _______________________________________________________________
Traceback (most recent call last):
  File "/awx_devel/awx/main/tests/unit/test_tasks.py", line 2216, in test_tower_source
    self.task.run(self.pk)
  File "/awx_devel/awx/main/tasks.py", line 601, in _wrapped
    return f(self, *args, **kwargs)
  File "/awx_devel/awx/main/tasks.py", line 1054, in run
    raise AwxTaskError.TaskError(instance, rc)
Exception: inventory_update None (error) encountered an error (rc=None), please see task stdout for details.
```

### after

```
_______________________________________________________________ TestInventoryUpdateCredentials.test_tower_source[True] _______________________________________________________________
Traceback (most recent call last):
  File "/awx_devel/awx/main/tests/unit/test_tasks.py", line 2218, in test_tower_source
    self.task.run(self.pk)
  File "/awx_devel/awx/main/tasks.py", line 601, in _wrapped
    return f(self, *args, **kwargs)
  File "/awx_devel/awx/main/tasks.py", line 1045, in run
    **extra_update_fields)
  File "/venv/awx/lib/python2.7/site-packages/mock/mock.py", line 1062, in __call__
    return _mock_self._mock_call(*args, **kwargs)
  File "/venv/awx/lib/python2.7/site-packages/mock/mock.py", line 1128, in _mock_call
    ret_val = effect(*args, **kwargs)
  File "/awx_devel/awx/main/tests/unit/test_tasks.py", line 250, in status_side_effect
    raise Exception('Task encountered error:\n{}'.format(kwargs['result_traceback']))
Exception: Task encountered error:
Traceback (most recent call last):
  File "/awx_devel/awx/main/tasks.py", line 944, in run
    env = self.build_env(instance, **kwargs)
  File "/awx_devel/awx/main/tasks.py", line 2054, in build_env
    env['TOWER_LICENSE_TYPE'] = get_licenser().validate()['license_type']
KeyError: 'license_type'
```